tests/queries/test_vouch_registry.py

```diff
@@ -0,0 +1,241 @@
+import unittest
+from dataclasses import dataclass
+
+from duneapi.types import DuneQuery, Network, QueryParameter, Address
+from duneapi.util import open_query
+
+from src.fetch.reward_targets import Vouch, parse_vouches, RECOGNIZED_BONDING_POOLS
+from tests.db.pg_client import (
+    ConnectionType,
+    DBRouter,
+    populate_from,
+)
+
+
+def pool_from(num: int) -> str:
+    return f"\\xb{num}"
+
+
+def sender_from(num: int) -> str:
+    return f"\\xf{num}"
+
+
+TEST_BONDING_POOLS = [
+    f"('{pool_from(i)}'::bytea, 'Pool {i}', '{sender_from(i)}'::bytea)"
+    for i in range(5)
+]
+
+
+@dataclass
+class MetaData:
+    schema: str
+    table: str
+    fields: list[str]
+
+
+# TODO - this can be part of the ORM i.e. the data classes modeling table records
+VOUCH_META = MetaData(
+    schema="cow_protocol",
+    table="VouchRegister_evt_Vouch",
+    fields=[
+        "solver",
+        '"bondingPool"',
+        '"cowRewardTarget"',
+        "sender",
+        "evt_index",
+        "evt_block_number",
+    ],
+)
+INVALIDATION_META = MetaData(
+    schema="cow_protocol",
+    table="VouchRegister_evt_InvalidateVouch",
+    fields=[
+        "solver",
+        '"bondingPool"',
+        "sender",
+        "evt_index",
+        "evt_block_number",
+    ],
+)
+
+
+def insert_query(meta: MetaData, events: list[str]) -> str:
```

Comment:
Maybe it would be better to call it insert_data?

Reply:
But the return type is the insertion query string...

Reply:
Perhaps you are confusing this with the other two methods explicitly named insert voices and insert invalidations.